source/AlarmWindow.cpp

```diff
@@ -50,72 +51,51 @@ AlarmWindow :: AlarmWindow (BRect frame, BHandler *handler)
+    char 	monthDefaultField[4];
+    char 	yearDefaultField[5];
+    char 	hourNow[6];
+	char	minNow[7];
 
 	// Initialize text fields with current system time values
-	sprintf(minuteDefaultField, "%02d", GetTime(0));
-	sprintf(hourDefaultField, "%02d", GetTime(1));
+	sprintf(minNow,"%02d", GetTime(0)+3);
 	sprintf(dayDefaultField, "%02d", GetTime(2));
+	sprintf(hourNow, "%02d", GetTime(1));
 	sprintf(monthDefaultField, "%02d", GetTime(3));
 	sprintf(yearDefaultField, "%d", GetTime(4));
 
 	// We allocate the view AlarmView and associate it to the AlarmWindow
-	frame.OffsetTo(B_ORIGIN);
-	fAlarmView = new AlarmView(frame,"AlarmView");
-	fAlarmView->SetViewColor(ui_color(B_PANEL_BACKGROUND_COLOR));
-	//AddChild(fAlarmView);
-
 	// Text fields for the data
-	hour = new BTextControl(BRect(10,10,100,35),   B_TRANSLATE("hour"),     B_TRANSLATE("hour:"), hourDefaultField , NULL);
-	minute = new BTextControl(BRect(110,10,210,35), B_TRANSLATE("minute"),  B_TRANSLATE("min:"), minuteDefaultField, NULL);
-	day = new BTextControl(BRect(10,50,100,75),   B_TRANSLATE("day"),  B_TRANSLATE("day:"), dayDefaultField, NULL);
-	month = new BTextControl(BRect(110,50,200,75), B_TRANSLATE("month"),    B_TRANSLATE("month:"), monthDefaultField, NULL);
-	year = new BTextControl(BRect(210,50,300,75),  B_TRANSLATE("year"),     B_TRANSLATE("year:"), yearDefaultField, NULL);
-
-	// Text field label: visible
-	hour -> SetDivider(hour->StringWidth("hour:") + 5);
-	minute -> SetDivider(minute->StringWidth("min:") + 5);
-	day -> SetDivider(day->StringWidth("day:") + 5);
-	month -> SetDivider(month->StringWidth("month:") + 5);
-	year -> SetDivider(year->StringWidth("year:") + 5);
+	hour = 		new BTextControl("hour",    B_TRANSLATE("hour:"), hourNow , NULL);
+	minute = 	new BTextControl("min",  B_TRANSLATE("min:"), minNow, NULL);
+	day = 		new BTextControl("day",  	B_TRANSLATE("day:"), dayDefaultField, NULL);
+	month = 	new BTextControl("month",   B_TRANSLATE("month:"), monthDefaultField, NULL);
+	year = 		new BTextControl("year",    B_TRANSLATE("year:"), yearDefaultField, NULL);
 
 	// Allocate the OK button
-	fButtonOk = new BButton (BRect(200,110,300,135),"ok", B_TRANSLATE("Ok"), new BMessage(BUTTON_ALARM_OK));
-	fButtonCancel = new BButton (BRect(320,110,420,135),"cancel",B_TRANSLATE("Cancel"),new BMessage(BUTTON_ALARM_CANCEL));
-
-	//BLayout
-	BLayoutBuilder::Group<>(this, B_VERTICAL, 0)
-		.AddGroup(B_HORIZONTAL)
-			.SetInsets(B_USE_DEFAULT_SPACING)
-			.Add(hour)
-			.Add(minute)
-			.AddGlue()
-		.End()
-		.AddGroup(B_HORIZONTAL)
-			.Add(day)
-			.Add(month)
-			.Add(year)
+	fButtonOk = new BButton ("ok", B_TRANSLATE("Ok"), new BMessage(BUTTON_ALARM_OK));
+	fButtonCancel = new BButton ("cancel",B_TRANSLATE("Cancel"),new BMessage(BUTTON_ALARM_CANCEL));
+
+	//thaflo, 2021, adding layout management
+	SetLayout(new BGroupLayout(B_VERTICAL));
+
+	BView* fTopView = new BGroupView(B_VERTICAL);
+
+	BLayoutBuilder::Group<>(fTopView, B_VERTICAL)
+		.SetInsets(5,5,5,5)
```

Comment:
haiku/headers/os/interface/InterfaceDefs.h has some constants to use for insets, e.g.:

```suggestion
		.SetInsets(B_USE_WINDOW_INSETS)
```